### PR TITLE
Run spellcheck against changed json file in PR

### PR DESCRIPTION
**Pull request by PhoenixHe-NV**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by NullMDR** [comment]
@yungezz @nschonni Please help to review this PR

**Comment by nschonni** [comment]
It LGTM, but you might want to just show it working by pushing a commit that fails the build. It could be rebased or reverted after

**Comment by NullMDR** [comment]
@nschonni Already done that test step. It works :)

**Review by raych1 — COMMENTED** [review]
Why don't we enable spellcheck for production pipeline in this PR?

**Comment by NullMDR** [comment]
@raych1 https://github.com/Azure/azure-rest-api-specs/pull/7104

File: scripts/spellcheck.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ }
+ if (result.stderr) {
+   console.error(result.stderr);
+ }
+ return result.code;
+}
+
+const main = async () => {
+  const config = cli.defaultConfig();
+  const pr = await devOps.createPullRequestProperties(config);
```
Comment (yungezz) [inline scripts/spellcheck.ts]: why need dependency on avocado? how about get changed files by github directly?

File: scripts/spellcheck.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      retCode = code;
+    }
+  }
+
+  return retCode;
+}
+
+main().then(retCode => {
+  if (retCode !== 0) {
+    console.error('Please fix the error or add words to custom-words.txt');
```
Comment (yungezz) [inline scripts/spellcheck.ts]: ./custom-words.txt?

File: specification/addons/resource-manager/Microsoft.Addons/preview/2017-05-15/examples/CanonicalSupportPlanTypes_Get.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
       "body": [
         {
           "enabled": true,
-          "supportPlanType": "standard",
+          "supportPlanType": "standad",
```
Comment (yungezz) [inline specification/addons/resource-manager/Microsoft.Addons/preview/2017-05-15/examples/CanonicalSupportPlanTypes_Get.json]: is this a real fix? or a spell error in swagger?

File: scripts/spellcheck.ts (hunk 1 above)
Comment (NullMDR) [reply scripts/spellcheck.ts]: That's kind of duplicated of logics. Get that changed file list require about 100 lines of code and I don't want to copy/paste that piece of code everywhere.

File: specification/addons/resource-manager/Microsoft.Addons/preview/2017-05-15/examples/CanonicalSupportPlanTypes_Get.json (hunk 3 above)
Comment (NullMDR) [reply specification/addons/resource-manager/Microsoft.Addons/preview/2017-05-15/examples/CanonicalSupportPlanTypes_Get.json]: To test the spellcheck output. Will revert back later.

**Comment by yungezz** [comment]
a more general question, for known text pattern like constants defined in UPPER_CASE, variables leading with $, like example in https://dev.azure.com/azure-sdk/public/_build/results?buildId=100800&view=logs&j=eca5a1cb-f12a-57d8-9254-390bd587c201&t=be275096-42fe-5449-ec4b-01c6f327d2b3&l=19, what's the preparation for them?

**Comment by NullMDR** [comment]
@yungezz The problem is not about uppercase or something. Spellcheck could recognize MAX_SIZE but not MAXSIZE unless we add MAXSIZE to custom-words.txt. And GOARCH is actually not a word. 
Service team need to maintain the custom-words.txt to add these words, but definitely we will avoid some spell error in swagger specs.

File: scripts/spellcheck.ts (hunk 2 above)
Comment (NullMDR) [reply scripts/spellcheck.ts]: Fixed

**Comment by NullMDR** [comment]
@yungezz Any more comments?

**Comment by NullMDR** [comment]
@yungezz Any concern for this one?

**Review by yungezz — APPROVED** [review]

**Pull request merged**